test/transforms.ts

```diff
@@ -101,3 +101,124 @@ test('abort signal', async () => {
     }
   `)
 })
+
+test('geojson', async () => {
+  const res = await postgrest
+    .from('shops')
+    .select()
+    .geojson()
+    .then((res) => res.data)
+  expect(res).toMatchInlineSnapshot(`
+    Object {
+      "features": Array [
+        Object {
+          "geometry": Object {
+            "coordinates": Array [
+              -71.10044,
+              42.373695,
+            ],
+            "type": "Point",
+          },
+          "properties": Object {
+            "address": "1369 Cambridge St",
+            "id": 1,
+          },
+          "type": "Feature",
+        },
+      ],
+      "type": "FeatureCollection",
+    }
+  `)
+})
+
+test('explain', async () => {
+  const res = await postgrest
+    .from('users')
+    .select()
+    .explain()
+    .then((res) => res.data)
+  expect(res).toMatchInlineSnapshot(`
+    Array [
+      Object {
+        "Plan": Object {
+          "Node Type": "Aggregate",
+          "Parallel Aware": false,
+          "Partial Mode": "Simple",
+          "Plan Rows": 1,
+          "Plan Width": 112,
+          "Plans": Array [
+            Object {
+              "Alias": "users",
+              "Node Type": "Seq Scan",
+              "Parallel Aware": false,
+              "Parent Relationship": "Outer",
+              "Plan Rows": 510,
+              "Plan Width": 132,
+              "Relation Name": "users",
+              "Startup Cost": 0,
+              "Total Cost": 15.1,
+            },
+          ],
+          "Startup Cost": 17.65,
+          "Strategy": "Plain",
+          "Total Cost": 17.68,
+        },
+      },
+    ]
+  `)
+})
+
+test('explain with options', async () => {
+  const res = await postgrest
+    .from('users')
+    .select()
+    .explain({ verbose: true, settings: true })
```

Comment:
Here I wanted to test adding `.single()`, like

```js
  const res = await postgrest
    .from('users')
    .select()
    .single()
    .explain()
```

But I got

```
test/transforms.ts:176:6 - error TS2339: Property 'explain' does not exist on type 'PromiseLike<PostgrestSingleResponse<any>>'.
```

This works when running on `node` though.

Any advice on how to make it work? (@soedirgo) 

---

This is necessary because the plan can change depending on the transform.`geojson()` or `csv()` change the generated query.

Reply:
`.single()` is meant to be called last, not followed by another method - it was the easiest way to implement the typings.

Given both `.single()` and `.geojson()` sets `Accept`, we don't need to support having both at the same time, no?

Reply:
saw the comment in the `.explain()` method - I think it's acceptable to not allow doing both for now, we can think about how to support both after launch week

Reply:
> .single() is meant to be called last, not followed by another method - it was the easiest way to implement the typings.

I see, I was trying to make `explain()` the last method to be called, it can be done after a `.csv()` or `.single()`.

> saw the comment in the .explain() method - I think it's acceptable to not allow doing both for now, we can think about how to support both after launch week

Ok, agree.